pyogrio/_io.pyx

```diff
@@ -195,6 +195,17 @@ cdef OGRLayerH get_ogr_layer(GDALDatasetH ogr_dataset, layer) except NULL:
     except CPLE_BaseError as exc:
         raise DataLayerError(str(exc))
 
+    # if the driver is OSM, we need to execute SQL to set the layer to read in
+    # order to read it properly
```

Comment:
Is this something to report to GDAL? Or is it expected we need to set this differently?

Reply:
`ogr2ogr` does this (which is how I figured out how to do this), so I think it is fair to say that GDAL knows about this and that it is intentional it is left for the user to call instead of being embedded in the driver.  It is something we could suggest be added to the documentation for the driver, however.